### PR TITLE
Clehner/enable stacking

### PR DESCRIPTION
**Pull request by lehnerchristian**
The idea of this PR is that the LineUpLayoutContainer uses uses the full width / height for each view. this PR should add the possibility that the views don't take the full width / height but that the views are only as wide / high as defined.

Related issue http://github.com/datavisyn/dTiles#53

### CODE REVIEW

File: src/assets/dnd/overflowing.svg [hunk 1]
```diff
@@ -0,0 +1,209 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!-- Created with Inkscape (http://www.inkscape.org/) -->
+
+<svg
```
Comment (sgratzl) [inline src/assets/dnd/overflowing.svg]: can you export the files as plain SVG (save as -> ...) such that all the inkscape metadata is removed and maybe even a manual cleanup afterwards

File: src/layout/styles/_dropper.scss [hunk 2]
```diff
@@ -79,4 +65,134 @@
...
+      &::after {
+        content: '';
+        display: block;
+        background-repeat: no-repeat;
+        background-position: 50%;
+        background-size: 5em;
+      }
+    }
+
+    .top-bottom,
```
Comment (sgratzl) [inline src/layout/styles/_dropper.scss]: please avoid to many css classes, especially with such a generic name e.g `center` I cannot predict if there will be some interference with other styles. In this setup I would go with, first, nth-child, last since you exactly know the setup already.

File: src/layout/internal/dropper.ts [hunk 3]
```diff
@@ -8,13 +8,20 @@ import SplitLayoutContainer from './SplitLayoutContainer';
 import {dropAble} from 'phovea_core/src/internal/dnd';
 import {EOrientation, IDropArea} from './interfaces';
 import {AParentLayoutContainer} from './AParentLayoutContainer';
+import LineUpLayoutContainer from './LineUpLayoutContainer';
```
Comment (sgratzl) [inline src/layout/internal/dropper.ts]: the lineup container doesn't put them next to each other but page wise. Is this change / option to be added in another PR?

Comment (lehnerchristian) [reply src/layout/internal/dropper.ts]: where should the widths of the different views be defined? or should they all get the same width defined somewhere in CSS?

File: src/layout/internal/LineUpLayoutContainer.ts [hunk 4]
```diff
@@ -1,16 +1,23 @@
 import {ILayoutContainer, ILayoutDump} from '../interfaces';
 import {EOrientation, IDropArea} from './interfaces';
-import {ALayoutContainer} from './ALayoutContainer';
+import {ALayoutContainer, ILayoutContainerOption} from './ALayoutContainer';
 import {ASequentialLayoutContainer, ISequentialLayoutContainerOptions, wrap} from './ASequentialLayoutContainer';
 
+export interface ILineUpLayoutContainer extends ISequentialLayoutContainerOptions {
+  stackLayout?: boolean;
```
Comment (sgratzl) [inline src/layout/internal/LineUpLayoutContainer.ts]: why optional, since already Partial is used later one?

File: src/layout/internal/LineUpLayoutContainer.ts [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
 
-export default class LineUpLayoutContainer extends ASequentialLayoutContainer<ISequentialLayoutContainerOptions> {
+export default class LineUpLayoutContainer extends ASequentialLayoutContainer<ILineUpLayoutContainer> {
   readonly minChildCount = 1;
   readonly type = 'lineup';
 
-  constructor(document: Document, options: Partial<ISequentialLayoutContainerOptions>, ...children: ILayoutContainer[]) {
-    super(document, options);
+  constructor(document: Document, options: Partial<ILineUpLayoutContainer>, ...children: ILayoutContainer[]) {
+    super(document, Object.assign({ stackLayout: false }, options));
```
Comment (sgratzl) [inline src/layout/internal/LineUpLayoutContainer.ts]: prefer overriding and extending the `defaultOptions` method

File: src/layout/internal/LineUpLayoutContainer.ts (hunk 4 above)
Comment (sgratzl) [reply src/layout/internal/LineUpLayoutContainer.ts]: and since this is a new option, please extends also `builder.ts` accordingly and the dumping and restoring methods

File: src/layout/internal/dropper.ts (hunk 3 above)
Comment (sgratzl) [reply src/layout/internal/dropper.ts]: the interface already has an optional `minSize` information which can be used. For the rest, it should be the "natural" width/height of each view.

File: src/layout/internal/AParentLayoutContainer.ts [hunk 6]
```diff
@@ -104,6 +104,7 @@ export abstract class AParentLayoutContainer<T extends ILayoutContainerOption> e
   }
 
   remove(child: ILayoutContainer) {
+    this.cleanup();
```
Comment (sgratzl) [inline src/layout/internal/AParentLayoutContainer.ts]: what is the difference to `takeDownChild`?

**Comment by thinkh** [comment]
@lehnerchristian Could you please briefly describe in your initial post what this PR does and/or link to an issue? Now it's not obvious to me what this PR does nor how to use it. Thanks.

**Comment by lehnerchristian** [comment]
@thinkh here's the issue: https://github.com/datavisyn/dTiles/issues/53
the idea is that the LineUpLayoutContainer uses uses the full width / height for each view. this PR should add the possibility that the views don't take the full width / height but that the views are only as wide / high as defined

**Comment by thinkh** [comment]
@lehnerchristian Thanks for the info. 👍
